### PR TITLE
Duosi/fix/refactor team page

### PR DESCRIPTION
**Pull request by Duosi-Dai**
The about/team page now display only PGs from each group by default. A button is added for each group to show more or show less members (OTs).

**New created components: TeamUpdate & TeamUpdateList** 
**TeamUpdate :** 
- Handles the conditional display of members based on their roles.
- Manages the state of the "Show more" or "Show less" button.

**TeamUpdateList:**
- Renders the general personal profile layout.
- Called within the **TeamUpdate** component to display individual member profiles.

**Updated: about/team page** : Integrated the **TeamUpdate** component to render the team members.

TeamPageRefactor

### CODE REVIEW

File: src/app/about/_components/TeamUpdateList.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { Person } from "@/components/shared/hooks/api/useOrganization"
+import { PersonIcon } from "@radix-ui/react-icons"
+import { LinkedinIcon, MailIcon } from "lucide-react"
+import Image from "next/image"
+import Link from "next/link"
+
+const TeamUpdateList = (person: Person) => {
+	return (
+		<>
+			{" "}
```
Comment (hampfh) [inline src/app/about/_components/TeamUpdateList.tsx]: why is this needed?

File: src/app/about/_components/TeamUpdate.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	const operationTeam = organization.people.filter(person =>
+		person.role.includes("Operation Team")
+	)
+
+	return (
+		<div key={organization.name} className="mt-16">
+			<h2 className="font-bebas-neue text-3xl">{organization.name}</h2>
+			<div className="mt-5 flex flex-wrap items-start justify-center gap-6 md:justify-start">
+				{projectGroup.map(person => (
+					<>
```
Comment (hampfh) [inline src/app/about/_components/TeamUpdate.tsx]: No need to use fragment here

File: src/app/about/_components/TeamUpdate.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+								id={person.id}
+								name={person.name}
+								email={person.email}
+								picture={person.picture}
+								linkedin_url={person.linkedin_url}
+								programme={person.programme}
+								role={person.role}
+							/>
+						))
+					: null}
```
Comment (hampfh) [inline src/app/about/_components/TeamUpdate.tsx]: can be shortened by using `&&` instead of `?`:

```ts
{updateTeam && operationTeam.map(person => (
  <TeamUpdateList
  key={person.id}
  id={person.id}
  name={person.name}
  email={person.email}
  picture={person.picture}
  linkedin_url={person.linkedin_url}
  programme={person.programme}
  role={person.role}
/>
))
```

File: src/app/about/_components/TeamUpdate.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+					? operationTeam.map(person => (
+							<TeamUpdateList
+								key={person.id}
+								id={person.id}
+								name={person.name}
+								email={person.email}
+								picture={person.picture}
+								linkedin_url={person.linkedin_url}
+								programme={person.programme}
+								role={person.role}
```
Comment (hampfh) [inline src/app/about/_components/TeamUpdate.tsx]: you don't need to spread the attributes here, could just pass it in like this:
`person={person}`, then change the TeamUpdateList component signature to:
`function TeamUpdateList({ person }: { person: Person }) {`

File: src/app/about/_components/TeamUpdate.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+		setUpdateTeam(prevState => !prevState)
+	}
+
+	const projectGroup = organization.people.filter(
+		person =>
+			person.role.includes("Project Group") ||
+			person.role.includes("Project Manager")
+	)
+	const operationTeam = organization.people.filter(person =>
+		person.role.includes("Operation Team")
```
Comment (hampfh) [inline src/app/about/_components/TeamUpdate.tsx]: here I have a tip, I would suggest that you lowercase the role and the match. For example if they call it `Project group` then this won't work. So what you could do is
```ts
person.role.toLowerCase().includes("project group")
```
then it doesn't matter how it is cased

File: src/app/about/team/page.tsx [hunk 6]
```diff
@@ -24,54 +21,9 @@ export default async function TeamPage() {
...
-														<LinkedinIcon className="inline-block aspect-square w-5" />
-													</Link>
-												)}
-											</div>
-										</div>
-									</div>
-								))}
-							</div>
-						</div>
+						<>
```
Comment (hampfh) [inline src/app/about/team/page.tsx]: no need for fragmentes

**Review by hampfh — COMMENTED** [review]
I like the way you have separated the logic into files and I think the functionality works great 👍 

I've left some comments on more code specific things. I think we can work with the naming a little more of the state-variable `updateTeam` and the components `TeamUpdate` and `TeamUpdateList`.

* the state variable `updateTeam` maybe could be renamed to `showOperationTeam`

* The component `TeamUpdate` maybe could be called something like RoleList or OrganizationList

* The TeamUpdateList is just a person so maybe call it PersonCard

**Review by AugustPaulsrud — APPROVED** [review]
Good work! I appreciate that this not only solves the immediate issue, but also the future issue of filtering away host. I approve the code but will also have some comments on things you could change to make the readability better before merging.

-UpdateTeam use state could become clearer if it was called something like “showOTs”, then what a true or a false value is would become more obvious.
-TeamUppdateList creates one card or item for the list, so it could be called teamMemeberItem, teamMemeberCard or something similar. But the teamUppdate file could be called teamList or something similar as that does create a list of team members.

These are some minor things, but in general it is good code with good structure and appropriately divided into multiple files.

File: src/app/about/_components/TeamUpdateList.tsx (hunk 1 above)
Comment (Duosi-Dai) [reply src/app/about/_components/TeamUpdateList.tsx]: Thanks for pointing out!

Deleted.

File: src/app/about/_components/TeamUpdate.tsx (hunk 3 above)
Comment (Duosi-Dai) [reply src/app/about/_components/TeamUpdate.tsx]: Thanks for advice, changed to `&&`

File: src/app/about/_components/TeamUpdate.tsx (hunk 4 above)
Comment (Duosi-Dai) [reply src/app/about/_components/TeamUpdate.tsx]: really helpful suggestion, thanks! 

Have changed

File: src/app/about/_components/TeamUpdate.tsx (hunk 5 above)
Comment (Duosi-Dai) [reply src/app/about/_components/TeamUpdate.tsx]: Have changed

**Comment by Duosi-Dai** [comment]
Hi,

Thanks for overall comments and suggestions. 

The new created components have changed to `OrganizationList` and `PersonCard`, state-variable has changed to name `showOTs`. Some small modifications have also changed based on the comments.
